File: 1). DSA/1). Data Structures/01). Array/C++/_005)_Left_Rotate_Array_by_1.cpp
#include <iostream>
#include <cmath>
using namespace std;

void lRotateOne(int arr[], int n)
{
	int temp = arr[0];
	for(int i = 1; i < n; i++)
	{
		arr[i - 1] = arr[i];
	}
	arr[n - 1] = temp;
}

int main()
{
       int arr[] = {1, 2, 3, 4, 5}, n = 5;
       cout<<"Before Left Rotation"<<endl;
       for(int i = 0; i < n; i++)
       {
       		cout<<arr[i]<<" ";
       }
       cout<<endl;
       lRotateOne(arr, n);
       cout<<"After '1' Left Rotation"<<endl;
       for(int i = 0; i < n; i++)
       {
       		cout<<arr[i]<<" ";
       }
}

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/Dart/_005)_Left_Rotate_Array_by_1.dart
void leftRotateOne(arr) {
  var temp = arr[0];
  for (int i = 1; i < arr.length; i++) {
    arr[i - 1] = arr[i];
  }
  arr[arr.length - 1] = temp;
}

void main() {
  var arr = [1, 2, 3, 4, 5];
  print('Before Left Rotate: ${arr}');
  leftRotateOne(arr);
  print('After Left Rotate: ${arr}');
}
